fpdf/table.py

```diff
@@ -260,7 +281,17 @@ def get_cell_border(self, i, j):
             return 1
         if self._borders_layout == TableBordersLayout.NONE:
             return 0
-        columns_count = max(row.cols_count for row in self.rows)
+        # In order to figure out whether a cell is the rightmost in its row, we
+        # need to access `colspan` of the corresponding `Cell` instance.
+        # That is because the `j` argument received from `_render_table_cell`
+        # is incremented with the colspan from cell to cell
+        columns_count = len(self.rows[i].cells)
+        colidx = 0
+        column_indexes = [colidx]
+        for jj in range(columns_count - 1):
+            colidx += self.rows[i].cells[jj].colspan
+            column_indexes.append(colidx)
+        rightmost_column = j == column_indexes[-1]
```

Comment:
```suggestion
        is_rightmost_column = j == self.rows[i].column_indices[-1]
```
I think this computation would be better placed inside a dedicated method or `@property` of the `Row` class.

Reply:
Nice suggestion, makes the code a lot cleaner 😄